tests/core/buffer.c

```diff
@@ -1204,3 +1204,37 @@ void test_core_buffer__dont_grow_borrowed(void)
 
 	cl_git_fail_with(GIT_EINVALID, git_buf_grow(&buf, 1024));
 }
+
+void test_core_buffer__dont_hit_infinite_loop_when_resizing(void)
+{
+	git_buf buf = GIT_BUF_INIT;
+
+	cl_git_pass(git_buf_puts(&buf, "foobar"));
+	cl_git_fail(git_buf_try_grow(&buf, SIZE_MAX, true));
```

Comment:
It looks like macOS isn't behaving the way you want.  I think that you're relying on allocation semantics here and that you may need to substitute an intentionally failing allocator here that asserts that it ends up with `SIZE_MAX` and fails appropriately.

Reply:
Yeah, I expected this to happen but hoped it wouldn't, so I just gave it a try. I've fixed the tests to not rely on allocation semantics